File: src/main/java/org/example/infraestructure/persistence/entity/EstudianteEntity.java
package org.example.infraestructure.persistence.entity;

import org.example.shared.enums.Carrera;

public class EstudianteEntity {

    private String matricula;
    private String nombre;
    private String apellidos;
    private int edad;
    private Carrera carrera;
    private String grupo;

    public EstudianteEntity() {}

    public EstudianteEntity(String matricula, String nombre, String apellidos, int edad, Carrera carrera, String grupo) {
        this.matricula = matricula;
        this.nombre = nombre;
        this.apellidos = apellidos;
        this.edad = edad;
        this.carrera = carrera;
        this.grupo = grupo;
    }

    public String getMatricula() {
        return matricula;
    }

    public void setMatricula(String matricula) {
        this.matricula = matricula;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellidos() {
        return apellidos;
    }

    public void setApellidos(String apellidos) {
        this.apellidos = apellidos;
    }

    public int getEdad() {
        return edad;
    }

    public void setEdad(int edad) {
        this.edad = edad;
    }

    public Carrera getCarrera() {
        return carrera;
    }

    public void setCarrera(Carrera carrera) {
        this.carrera = carrera;
    }

    public String getGrupo() {
        return grupo;
    }

    public void setGrupo(String grupo) {
        this.grupo = grupo;
    }

}

File: src/main/kotlin/com/practica/crean/arquitecture/infraestructure/persistence/entity/EstudianteEntity.kt
package com.example.demo.infraestructure.persistence.entity

import com.example.demo.shared.enums.Carrera
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "estudiantes")
/*
* Las entidades en Jakarta (Jakarta Persistence API) son clases mapeadas
* a tablas de bases de datos mediante anotaciones, representando datos
* persistentes en el sistema
* */
data class EstudianteEntity(

    @Id
    var matricula: String,
    var nombre: String,
    var apellidos: String,
    var edad: Int,
    var carrera: Carrera,
    var grupo: String

)